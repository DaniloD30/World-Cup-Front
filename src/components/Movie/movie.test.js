import React from "react";
import { shallow, mount, render } from "enzyme";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
import { BrowserRouter } from "react-router-dom";
import Index from "./Movie";

const mockStore = configureMockStore([thunk]);
const store = mockStore();

const handleListSelectMovies = () => {
  // const movieL = movieList.find((obj) => obj.id === movie.id);
  // if (movieL === undefined) {
  //   if (movieList.length < 8) {
  //     setMoviesList(movieList.concat(movie));
  //   }
  // }
};

const handleDelete = () => {
  // setMoviesList(
  //   movieList.filter((appointment) => appointment.id !== movie.id)
  // );
};
describe("Movie ", () => {
  beforeEach(() => {
    // Runs before each test in the suite
    Object.defineProperty(window, "matchMedia", {
      writable: true,
      value: jest.fn().mockImplementation((query) => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // deprecated
        removeListener: jest.fn(), // deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });
    store.clearActions();
  });

  it('it should render correctly with store in "debug" mode', () => {
    const store = mockStore({
      movie: {
        movies: [],
      },
    });

    const wrapper = shallow(
      <BrowserRouter>
        <Provider store={store} debug>
          <Index
            title="{item.titulo}"
            year="{item.ano}"
            handleList={handleListSelectMovies}
            handleDel={handleDelete}
            movie="{item}"
          />
        </Provider>
      </BrowserRouter>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("it should render correctly without store", () => {
    const store = mockStore({});

    const wrapper = mount(
      <BrowserRouter>
        <Provider store={store}>
          <Index
            title="{item.titulo}"
            year="{item.ano}"
            handleList={handleListSelectMovies}
            handleDel={handleDelete}
            movie="{item}"
          />
        </Provider>
      </BrowserRouter>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
