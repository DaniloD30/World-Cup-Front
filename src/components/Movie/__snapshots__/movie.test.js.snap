// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie  it should render correctly with store in "debug" mode 1`] = `ShallowWrapper {}`;

exports[`Movie  it should render correctly without store 1`] = `ReactWrapper {}`;
